test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -790,17 +789,12 @@ protected static void checkStaticState() throws Exception {
     }
 
     /**
-     * Assert that a leak was detected, also remove the leak from the list of detected leaks
-     * so the test won't fail for that specific leak.
-     *
-     * @param expectedPattern A pattern that matches the detected leak's exception
+     * Assert that at least one leak was detected, also clear the list of detected leaks
+     * so the test won't fail for leaks detected up until this point.
      */
-    protected static void assertLeakDetected(String expectedPattern) {
+    protected static void assertLeakDetected() {
         synchronized (loggedLeaks) {
-            assertTrue(
-                "No leak detected matching the pattern: " + expectedPattern,
-                loggedLeaks.removeIf(leakText -> Pattern.matches(expectedPattern, leakText))
-            );
+            assertTrue("No leaks have been detected", loggedLeaks.removeIf(leakText -> true));
```

Comment:
nit: seems a little too sneaky, I think I'd prefer a more explicit `assertFalse(empty())` and then `clear()` similar to the bottom of `assertStaticState()` above.